### ISSUE TITLE
Components: increase the helpText.fontSize

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#69467]
### Description

Splitting this out from https://github.com/WordPress/gutenberg/pull/69451#issuecomment-2703107519

In https://github.com/WordPress/gutenberg/pull/69451 the Visibility panel in the pre-publish checks has been refactored and now uses `RadioControl` which uses a 12 pixels font size for the help text.

https://github.com/WordPress/gutenberg/blob/997934d84d47df135421c85439db5559340a81b7/packages/components/src/utils/font-values.js#L6

The previous font size was 13 pixels. As such, the help text is now _smaller_ and it's actually a reduced legibility level.

Overall, I'd think there should't be any text in the editor that uses a font size smaller than 13 pixels.
12 pixels is really too small in 2025. I'd even consider to raise any minimum font size to at least 14 pixels but for now 13px would be a good start.

Of course, I do understand the help text must have, in a way, slightly less prominence. However, typography hierarchy doesn't necessarily need to use the font size. It can be provided by color, spacing, white space and the like. 

CC @WordPress/gutenberg-components and @WordPress/gutenberg-design 

### Step-by-step reproduction instructions

N/A

Screenshot to illustrate an example of help text with a 12 pixels font size (on the right) compared with the previous 13 pixels font size (on the left). Please ignore the different alignments / spacings and only focus on the help text font size.

![Image](https://github.com/user-attachments/assets/a36eb28b-ef67-4bc0-8465-de32187231f4)

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by Mamaduka** [comment]
Thanks for creating an issue, @afercia!

**Comment by ciampo** [comment]
Nothing to say against larger font size in general, although we should be aware of the implications of changing the variable:

- there may be design implications (especially about type hierarchy and spacing) that may be considered a design regressions, and would therefore require other design adjustments to allow for the font size change
- let's make sure to update, in case, any corresponding sass variables across the repo (likely in the base scss styles files)
- let's make sure to check the consequences everywhere the variable is used

**Comment by afercia** [comment]
I'd agree on all points and this is definitely an effort I'd love to see establishing new typography system guidelines across all WordPress.
Cc @WordPress/gutenberg-design

**Comment by hanneslsm** [comment]
Related: https://github.com/WordPress/gutenberg/issues/64340

Just to clarify, the issue is that `12px` is too small we should agree on a minimum font size - but it's not that we're talking about  the issue with `pixels` as font size instead of `rem`?
Where is the info from that 12px is too small? 

As a comparison, the Github interface is using a lot of 12px text: https://primer.style/product/primitives/typography/

<img width="397" alt="Image" src="https://github.com/user-attachments/assets/55f93413-c0c1-4724-a3d5-521a04d18791" />

<img width="942" alt="Image" src="https://github.com/user-attachments/assets/2182c98b-486b-40de-9f00-7a3dce99521c" />

**Comment by carolinan** [comment]
GitHub is not comparable, it is not an editing tool.

WCAG does not specify a font size recommendation or requirement besides the contrast ratio for "large text".
But there is ongoing research for text appearance in WCAG 3.0 here: https://github.com/w3c/wcag3/issues/266
